src/program/program.controller.ts

```diff
@@ -35,12 +51,15 @@ export class ProgramController {
     @Param('programContentId') programContentId: string,
   ) {
     const { memberId } = request.query;
-    const { role, permissions } = member;
 
-    const extraAllowPermission = ['PROGRAM_NORMAL'].find((e) => permissions.includes(e));
+    const programContent = await this.programService.getProgramContentById(programContentId);
+    const loginToTrialProgramContent = await this.programService.getLoginToTrialProgramContent(programContentId);
```

Comment:
add try catch to catch error
if error cause , should throw api exception in controller layer